etest/docker/image.py

```diff
@@ -24,7 +24,11 @@ def pull(image_name):
         if error.response.status_code != 404:
             raise error
 
-    repository, tag = image_name.split(":")
+    try:
+        repository, tag = image_name.split(":")
+    except ValueError:
+        repository = image_name
+        tag = None
```

Comment:
This feels like it should be handled by split but maybe a check for colon then just doing two different assignments?

repository, tag = image_name.split(":") if ":" in image_name else image_name, None

Reply:
this was added because .split() gave errors on tag-less images.

Reply:
Yeah, that makes sense.  I wonder if adding a tag makes more sense but docker does have its defaults and it's probably better to use theirs than ours.  This simply seems really heavy handed to me but Python isn't a beautiful language so this might be reasonably idiomatic.

Reply:
I did this so we could directly pull the qemu intepreter from dockerhub without retagging. It might be a little bit ugly but it does simplify things a lot.

Reply:
Ah, in that case it should be another argument so you can control that as the user without having strange code here.  Sanitise on ingress and all that. ;)

Reply:
I dont think thats really needed, since theres only 1 qemu interpreter container that you can use. Ive done some testing and it seems like adding :latest works too, so I did that and removed this part.